web/src/connectors/task-annotator-connector/task-annotator-context.tsx

```diff
@@ -252,39 +231,12 @@ export const TaskAnnotatorContextProvider: React.FC<ProviderProps> = ({
 
     const { notifyError } = useNotifications();
 
-    let task: Task | undefined;
-    let isTaskLoading: boolean = false;
-    let refetchTask: (
-        options?: (RefetchOptions & RefetchQueryFilters<Task>) | undefined
-    ) => Promise<QueryObserverResult<Task, unknown>>;
-    if (taskId) {
-        const result = useTaskById({ taskId }, {});
-        task = result.data;
-        isTaskLoading = result.isLoading;
-        refetchTask = result.refetch;
-    }
-
-    const getJobId = (): number | undefined => (task ? task.job.id : jobId);
-
-    const getFileId = (): number | undefined => (task ? task.file.id : fileMetaInfo?.id);
-    const { isOwner, taskUsers } = useTaskUsers(task);
-
+    const { data: task, isLoading: isTaskLoading, refetch: refetchTask } = useTaskById({ taskId });
```

Comment:
maybe should be enabled only if taskId?

Reply:
it's already enabled only if taskId is exists
```
export const useTaskById: QueryHookType<{ taskId?: number }, Task> = ({ taskId }) =>
    useQuery(
        ['task', taskId],
        async () =>
            useBadgerFetch<ApiTask>({
                url: `${namespace}/tasks/${taskId}`,
                method: 'get'
            })(),
        { select: mapTaskFromApi, enabled: Boolean(taskId) }
    );
    ```

Reply:
sorry, missed it